src/xma/src/xmaapi/xmaapi.cpp

```diff
@@ -456,56 +382,37 @@ int32_t xma_initialize(XmaXclbinParameter *devXclbins, int32_t num_parms)
         return XMA_ERROR;
     }
 
-    if (xrt_core::config::get_xma_mode2()) {
-        xma_core::utils::xma_enable_mode2();
+    int32_t exec_mode = xrt_core::config::get_xma_exec_mode();
```

Comment:
exec_mode should be an enumeration, especially when using it in a switch statement.